src/main/java/com/songr/songr/controllers/AppController.java

```diff
@@ -54,4 +60,16 @@ public RedirectView addAlbumFromForm(String title, String artist, Integer songCo
 
     return new RedirectView("/");
     }
+
+    @GetMapping("/albums/{id}")
+    public String getOneAlbum(Model model, @PathVariable Long id){
+        Album album = albumRepository.getById(id);
+        List<Song> songs = songRepository.findAll();
```

Comment:
Here, you're getting all the songs from your repository and listing them on the page, regardless of which album they're on. Instead, look at your Album model. Every time you create a new song, it gets added to the albums `songs` list. Since you're already retrieving the album, is there a way you could access its `songs` and pass that to Thymeleaf instead?